cmd/upgrade/operatorroles/cmd.go

```diff
@@ -278,20 +285,19 @@ func handleModeFlag(cmd *cobra.Command, skipInteractive bool, mode string, err e
 	return mode, err
 }
 
-func upgradeOperatorPolicies(mode string, r *rosa.Runtime,
-	prefix string, isAccountRoleUpgradeNeed bool, policies map[string]string, env string,
-	defaultPolicyVersion string, credRequests map[string]*cmv1.STSOperator, cluster *cmv1.Cluster) error {
-	path, err := aws.GetRolePath(cluster.AWS().STS().RoleARN())
-	if err != nil {
-		return err
-	}
+func upgradeOperatorPolicies(isProgrammaticallyCalled bool, mode string, r *rosa.Runtime,
+	prefix string, isAccountRoleUpgradeNeed bool, policies map[string]string, env string, defaultPolicyVersion string,
+	credRequests map[string]*cmv1.STSOperator, cluster *cmv1.Cluster, policyPath string) error {
 	switch mode {
 	case aws.ModeAuto:
 		if !confirm.Prompt(true, "Upgrade the operator role policy to version %s?", defaultPolicyVersion) {
+			if isProgrammaticallyCalled {
+				return r.Reporter.Errorf("Operator roles need to be upgraded to proceed")
+			}
```

Comment:
Why is this here? Why are we blocking the user from upgrading the roles?

Reply:
This was the behavior in account roles. When called from cluster upgrade (isProgramatically is true) then account roles upgrade prompts the usrr and returns error if user does not confirm